### ISSUE TITLE
Kali installation "distutils.spawn" not found

### ISSUE THREAD

**Issue by TheThingGoesSkra** [securestate/king-phisher#407]
## Issue Description
Hello, I experience an issue when I tried to install King-phisher on Kali using the install.sh script on a totally fresh environment. 

## Environment Details

Host OS: Kali 2019.1a from osboxes.org
King Phisher Version: 1.5
Pipenv version: 2018.11.26
## Error Details / Stack Trace

⠹ Creating virtual environment...Running virtualenv with interpreter /usr/bin/python3.6m

✘ Failed creating virtual environment 
[pipenv.exceptions.VirtualenvCreationException]:   File "/usr/local/lib/python3.7/dist-packages/pipenv/vendor/click/decorators.py", line 17, in new_func
[pipenv.exceptions.VirtualenvCreationException]:       return f(get_current_context(), *args, **kwargs)
[pipenv.exceptions.VirtualenvCreationException]:   File "/usr/local/lib/python3.7/dist-packages/pipenv/cli/command.py", line 208, in cli
[pipenv.exceptions.VirtualenvCreationException]:       clear=state.clear,
[pipenv.exceptions.VirtualenvCreationException]:   File "/usr/local/lib/python3.7/dist-packages/pipenv/core.py", line 574, in ensure_project
[pipenv.exceptions.VirtualenvCreationException]:       pypi_mirror=pypi_mirror,
[pipenv.exceptions.VirtualenvCreationException]:   File "/usr/local/lib/python3.7/dist-packages/pipenv/core.py", line 506, in ensure_virtualenv
[pipenv.exceptions.VirtualenvCreationException]:       python=python, site_packages=site_packages, pypi_mirror=pypi_mirror
[pipenv.exceptions.VirtualenvCreationException]:   File "/usr/local/lib/python3.7/dist-packages/pipenv/core.py", line 935, in do_create_virtualenv
[pipenv.exceptions.VirtualenvCreationException]:       extra=[crayons.blue("{0}".format(c.err)),]
[pipenv.exceptions.VirtualenvCreationException]: Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/virtualenv.py", line 24, in <module>
    import distutils.spawn
ModuleNotFoundError: No module named 'distutils.spawn'

Failed to create virtual environment.
pipenv encountered the following error:
ERROR    failed to install the pipenv environment
ERROR    failed to remove the incomplete .venv directory
Traceback (most recent call last):
  File "/opt/king-phisher/king_phisher/startup.py", line 188, in pipenv_entry
    shutil.rmtree(os.path.join(target_directory, '.venv'))
  File "/usr/lib/python3.7/shutil.py", line 485, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib/python3.7/shutil.py", line 483, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/opt/king-phisher/.venv'

**Comment by zeroSteiner** [comment]
Did you really mean that you're running version 1.5 or is that a type and you mean 1.15? Also can you please provide entire command you ran and all of the output? If it's alot you can post it to pastebin or something.

**Comment by TheThingGoesSkra** [comment]
In fact, I was running the version 1.15. But why ? I downloaded the last version directly from the github page.

 You have all I have done on the system once the vm setup in this Pastebin : https://pastebin.com/8x5VRrmJ

Thanks you very much !

**Comment by zeroSteiner** [comment]
Can you try `sudo apt-get install python3-distutils`. It looks like distutils isn't properly setup.

**Comment by TheThingGoesSkra** [comment]
It's done. Nothing happened I have already the last version... I have try in my python console with python3.7, 3.6 and 2, distutils is available but not distutils.spawn, distutils.util.spawn and distutils.core.spawn too.

**Comment by TheThingGoesSkra** [comment]
I beg your pardon, In fact, I also have another problem... I was developing extra-functionalities on the KingPhisherServer Version : 1.13.2 and I had a problem with the "King-phisher-client.ui" file. I have deleted some parts of the code.

I would like to know if you have please, this version 1.13.2 of this file in order to help me to repair my mistake ?

**Comment by zeroSteiner** [comment]
Well you don't need the UI file for the server. That's all the client graphics data.

The other issue looks related to [this issue](https://github.com/pypa/pipenv/issues/2922) where they state that it's due to a broken distutils installation.

**Comment by TheThingGoesSkra** [comment]
Ok, I solved my problem with the client UI version 1.13.2.

I still trying to install the new version of king phisher on Kali Linux but I keep the same errors above. The solutions proposed in the issue you've spoke about don't mitigate it.

**Comment by zeroSteiner** [comment]
I'm going to need some time to try and reproduce this issue since I can't think of what it would be off the top of my head.

**Comment by TheThingGoesSkra** [comment]
No problem! It was principally to prevent you. Thank you.

**Comment by zeroSteiner** [comment]
I just tried this and I can't reproduce this. I installed a fresh copy of Kali 2019.3 AMD64 Light, installed updates and then ran the install script.

**Comment by True-Demon** [comment]
I can't help but notice that the version of Kali (2019.1a) you're using was from a non-official source. (osboxes.org)

Despite reporting the same version as the official kali repository, their SHA hashes do not match, indicating there were changes made to that OS version that we have no knowledge of. I would strongly recommend retrying this installation from a fresh Kali image from the official Offensive Security downloads page: https://www.kali.org/downloads/

There's no telling what changes were made to that image and whether distutils or any other component of the OS that could have caused the problems you're experiencing.

If the sole purpose of this server is to act as a King-Phisher server, I would suggest installing it on an Ubuntu 18.04 or Debian 9 server, as these are stable, reliable, and are not nearly as weighed down as the Kali OS. If you _need_ Kali, then definitely install it from the official sources.

TLDR: Don't touch that image from osboxes.org. You don't know where it's been. ;)

**Comment by TheThingGoesSkra** [comment]
Thanks you very much ! I'm sorry for the disturbance... I will try to install it from the official sources.

**Comment by zeroSteiner** [comment]
For the record, I've heard other reports of folks being able to reproduce this issue but they have also mentioned that it is inconsistent. There may yet be a larger issue here.
